core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -820,6 +823,37 @@ class TransactionsTest extends IntegrationTestHarness {
     assertEquals((initialProducerEpoch + 1).toShort, producerStateEntry.producerEpoch)
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testTransactionsWithCompression(quorum: String): Unit = {
+    val numRecords = 50
+    val numProducersWithCompression = 5
+    val numTransactions = 40
+    val transactionalCompressionProducers = Buffer[KafkaProducer[Array[Byte], Array[Byte]]]()
+
+    for (i <- 0 until numProducersWithCompression) {
+      transactionalCompressionProducers += createTransactionalProducer("transactional-compression-producer-" + i.toString,  compressionType = "snappy")
+    }
+
+    // KAFKA-15653 is triggered more easily with replication factor 1
```

Comment:
This made sense to me before, but I'm trying to remember why. 😅 

Is it possible that when we are building and sending the response we could receive the callback request and that means it will be handled on a different thread? Vs when acks=all, we free the thread after sending out the verification, so it can be available to receive the callback?

Reply:
Not sure I follow that. Whether there are any replicas or not, wouldn't the request thread will be freed after sending out the verification? One way I could see the case being more likely is if we added additional request threads. I don't think there's any affinity in the callback to the original request thread, is there? So more available request threads probably means a greater chance a separate thread would handle the callback. Does that make sense?

Reply:
Sorry I think I got confused about when the response is being sent. I will say that adding more request threads didn't work and I had to lower them to get it to reproduce frequently?

I can try it the other way around, but iirc, there were a lot of combinations that were not as consistent as it is now.

Reply:
My take is that this test should probably just test compression of transactional data with the default settings. I don't think it necessarily needs to hit the case from KAFKA-15653. It seems like it doesn't do so reliably anyway. Are there lower level tests where we can validate callback safety?

Reply:
I didn't see it fail much at all with the default settings. It was like once in 60 times or less which was not super useful. There are lower level tests for the callback using the correct requestLocal.

Reply:
Yeah, what I'm saying is that this test seems more useful as a general validation of transactional produce with compression. From that perspective, the default settings are the most useful to test. For KAFKA-15653, the lower level tests seem sufficient.

Reply:
Interestingly when I put it back to the typical settings it failed consistently 🤦‍♀️ 
I guess I tricked myself into thinking i needed the other settings.

Reply:
I set back the default settings and it seems to trigger consistently again anyway. Maybe I just tricked myself into thinking I needed the other configs.